### PR TITLE
Enable binary data type TCK tests

### PR DESCRIPTION
**Pull request by aguibert**
For issue #499

### CODE REVIEW

File: vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/DB2ParamDesc.java [hunk 1]
```diff
@@ -40,12 +43,29 @@ public String prepare(TupleInternal values) {
...
+        case ClientTypes.BOOLEAN:
+        case ClientTypes.DECIMAL:
+        case ClientTypes.DOUBLE:
+        case ClientTypes.INTEGER:
+        case ClientTypes.REAL:
+        case ClientTypes.SMALLINT:
+        	return clazz == Numeric.class;
+    	}
+    	return false;
+    }
```
Comment (gjwatts) [inline vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/DB2ParamDesc.java]: Why not just add "return clazz == Numeric.class" to canConvert in ClientTypes?  Is there a case where ClientTypes is a BIT, CHAR or LONGVARCHAR and you want to return a Numeric class only if it is not one of those types?

File: vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/ColumnMetaData.java [hunk 2]
```diff
@@ -105,21 +107,48 @@ public void setColumnCount(final int numColumns) {
...
+    	if (sqlxName_ != null && i < sqlxName_.length && sqlxName_[i] != null)
+    		return sqlxName_[i];
     	// Otherwise use column names from SQLDOPTGRP
-    	if (sqlName_ != null && sqlName_.length > 0)
-    		return Collections.unmodifiableList(Stream.of(sqlName_).collect(Collectors.toList()));
-    	return Collections.emptyList();
+    	if (sqlName_ != null && i < sqlName_.length)
+    		return sqlName_[i];
+    	return null;
     }
```
Comment (gjwatts) [inline vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/ColumnMetaData.java]: Should we add a check to make sure the passed in i >= 0?  Or are we guaranteed a value that is 0+

File: vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ListTuple.java [hunk 3]
```diff
@@ -19,6 +19,7 @@
 
 import io.vertx.sqlclient.Tuple;
 
+import java.util.Iterator;
```
Comment (gjwatts) [inline vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/ListTuple.java]: Remove?

**Comment by gjwatts** [comment]
Should we update copyright dates in java files we change that are outside of DB2?

File: vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/DB2ParamDesc.java (hunk 1 above)
Comment (aguibert) [reply vertx-db2-client/src/main/java/io/vertx/db2client/impl/codec/DB2ParamDesc.java]: right, the other types like CHAR, LONGVARCHAR, (and in the future BLOB and CLOB) won't be convertable to Numeric

File: vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/ColumnMetaData.java (hunk 2 above)
Comment (aguibert) [reply vertx-db2-client/src/main/java/io/vertx/db2client/impl/drda/ColumnMetaData.java]: Good catch, I think this method will eventually be an SPI so I'll add the check

**Pull request merged**